File: jps-plugin/src/org/jetbrains/jps/incremental/scala/model/ProjectSettings.java
package org.jetbrains.jps.incremental.scala.model;

import org.jetbrains.jps.ModuleChunk;
import org.jetbrains.jps.model.JpsElement;

/**
 * @author Pavel Fatin
 */
public interface ProjectSettings extends JpsElement {
  public IncrementalityType getIncrementalityType();

  public CompilerSettings getCompilerSettings(ModuleChunk chunk);
}

File: jps-plugin/src/org/jetbrains/jps/incremental/scala/model/ProjectSettings.scala
package org.jetbrains.jps.incremental.scala
package model

import org.jetbrains.plugin.scala.compiler.{CompileOrder, IncrementalType}

/**
 * Nikolay.Tropin
 * 11/18/13
 */
trait ProjectSettings {

  def incrementalType: IncrementalType

  def compileOrder: CompileOrder
}